bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java

```diff
@@ -400,34 +415,25 @@ public boolean updateRelays(ShellySettingsStatus status) throws IOException {
      * @param profile ShellyDeviceProfile
      * @param status Last ShellySettingsStatus
      *
-     * @throws IOException
+     * @throws ShellyApiException
      */
-    @SuppressWarnings("null")
-    public boolean updateDimmers(ShellySettingsStatus orgStatus) throws IOException {
-        ShellyDeviceProfile profile = getProfile();
-
+    public boolean updateDimmers(ShellySettingsStatus orgStatus) throws ShellyApiException {
         boolean updated = false;
         if (profile.isDimmer) {
-            Validate.notNull(orgStatus, "orgStatus must not be null!");
-
             // We need to fixup the returned Json: The dimmer returns light[] element, which is ok, but it doesn't have
-            // the same structure as lights[] from Bulb and RGBW2. The tag gets replaced by dimmers[] so that Gson maps
-            // to a different structure (ShellyShortLight).
+            // the same structure as lights[] from Bulb,RGBW2 and Duo. The tag gets replaced by dimmers[] so that Gson
+            // maps to a different structure (ShellyShortLight).
             Gson gson = new Gson();
             ShellySettingsStatus dstatus = gson.fromJson(ShellyApiJsonDTO.fixDimmerJson(orgStatus.json),
                     ShellySettingsStatus.class);
             Validate.notNull(dstatus.dimmers, "dstatus.dimmers must not be null!");
-            Validate.notNull(dstatus.tmp, "dstatus.tmp must not be null!");
 
             logger.trace("{}: Updating {} dimmers(s)", thingName, dstatus.dimmers.size());
-
             int l = 0;
-            logger.trace("{}: Updating dimmers {}", thingName, dstatus.dimmers.size());
             for (ShellyShortLightStatus dimmer : dstatus.dimmers) {
                 Integer r = l + 1;
                 String groupName = profile.numRelays <= 1 ? CHANNEL_GROUP_DIMMER_CONTROL
                         : CHANNEL_GROUP_DIMMER_CONTROL + r.toString();
-                // updated |= updateChannel(groupName, CHANNEL_OUTPUT, getOnOff(dimmer.ison));
```

Comment:
Was this removed intentionally?

Reply:
Could you please re-check, my code doesn't show this line - strange

Reply:
ah', this was removed, because the Dimmer on/off is now moved in the general brightness handling (on/off is an integrated part of the UI control)